src/client/visualizations/line-chart/line-chart.tsx

```diff
@@ -323,26 +329,26 @@ export class LineChart extends BaseVisualization<LineChartState> {
     extentY: number[],
     scaleY: any
   ): JSX.Element {
-    const { clicker, essence } = this.props;
-    const { highlight } = essence;
+    const { essence } = this.props;
 
     const { containerYPosition, containerXPosition, scrollTop, dragRange } = this.state;
-    const { dragOnMeasure, scaleX, hoverRange } = this.state;
+    const { dragOnSeries, scaleX, hoverRange } = this.state;
 
-    if (highlight && !essence.highlightOn(series.measure.name)) return null;
+    if (this.hasHighlight() && !this.highlightOn(series.definition.key())) return null;
```

Comment:
highlightOn should be able to check hasHighlight internally

Reply:
But it is more complicated condition. It says if you have highlight and that highlight is different than X. `highlightOn` checks for existence of `highlight` object, but returns false if it is not present - You don't have highlight, so surely you don't have highlight on X.

Maybe extract to helper like `highlightedBySomeoneElse`

Reply:
complex condition extracted into named method +1000